src/services/oss/backend.rs

```diff
@@ -0,0 +1,977 @@
+        let p = build_abs_path(&self.root, path);
+
+        let url = format!("{}/?list-type=2", self.endpoint,);
+
+        let req = Request::get(&url)
+            .header(HOST, &self.host)
+            .header("delimiter", "/")
+            .header("prefix", p)
+            .header("continuation-token", token.unwrap_or_default())
+            .body(AsyncBody::Empty)
+            .map_err(|e| new_request_build_error(Operation::List, path, e))?;
+        Ok(req)
+    }
+
+    async fn oss_get_object(
+        &self,
+        path: &str,
+        offset: Option<u64>,
+        size: Option<u64>,
+    ) -> Result<Response<IncomingAsyncBody>> {
+        let mut req = self.oss_get_object_request(path, offset, size)?;
+
+        self.signer
+            .sign(&mut req)
+            .map_err(|e| new_request_sign_error(Operation::Read, path, e))?;
+        self.client
+            .send_async(req)
+            .await
+            .map_err(|e| new_request_send_error(Operation::Read, path, e))
+    }
+
+    async fn oss_head_object(&self, path: &str) -> Result<Response<IncomingAsyncBody>> {
+        let mut req = self.oss_head_object_request(path)?;
+
+        self.signer
+            .sign(&mut req)
+            .map_err(|e| new_request_sign_error(Operation::Stat, path, e))?;
+        self.client
+            .send_async(req)
+            .await
+            .map_err(|e| new_request_send_error(Operation::Stat, path, e))
+    }
+
+    async fn oss_put_object(
+        &self,
+        path: &str,
+        size: Option<u64>,
+        body: AsyncBody,
+    ) -> Result<Response<IncomingAsyncBody>> {
+        let mut req = self.oss_put_object_request(path, size, body)?;
+
+        self.signer
+            .sign(&mut req)
+            .map_err(|e| new_request_sign_error(Operation::Write, path, e))?;
+        self.client
+            .send_async(req)
+            .await
+            .map_err(|e| new_request_send_error(Operation::Write, path, e))
+    }
+
+    pub(super) async fn oss_list_object(
+        &self,
+        path: &str,
+        token: Option<String>,
+    ) -> Result<Response<IncomingAsyncBody>> {
+        let mut req = self.oss_list_object_request(path, token)?;
+
+        self.signer
+            .sign(&mut req)
+            .map_err(|e| new_request_sign_error(Operation::List, path, e))?;
+        self.client
+            .send_async(req)
+            .await
+            .map_err(|e| new_request_send_error(Operation::List, path, e))
+    }
+
+    async fn obs_delete_object(&self, path: &str) -> Result<Response<IncomingAsyncBody>> {
+        let mut req = self.oss_delete_object_request(path)?;
+        self.signer
+            .sign(&mut req)
+            .map_err(|e| new_request_sign_error(Operation::Delete, path, e))?;
+        self.client
+            .send_async(req)
+            .await
+            .map_err(|e| new_request_send_error(Operation::Delete, path, e))
+    }
+
+    async fn oss_initiate_multipart_upload(
```

Comment:
Don't try to finish all work in one PR. Split them one by one.

I prefer to ensure oss works (read/write/list) first and then add multipart uploads.